File: Data Structures & Algorithms/duplicate-integer/submission-1.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_set<int> uniqueVals;

        for (int i = 0; i < nums.size(); i++) {
            if (uniqueVals.count(nums[i])) {
                return true;
            }

            uniqueVals.insert(nums[i]);
        }

        return false;
    }
};

File: Data Structures & Algorithms/jump-game/submission-1.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        n = len(nums)
        dp = [False] * n

        dp[n - 1] = True

        for i in range(n - 2, -1, -1):
            for j in range(1, nums[i] + 1):
                if dp[i + j]:
                    dp[i] = True
                    break
        
        return dp[0]
